demo/api/src/warnings/warning-checker.console.ts

```diff
@@ -0,0 +1,125 @@
+import { Block, BlockInput, FlatBlocks } from "@comet/blocks-api";
+import { DiscoverService } from "@comet/cms-api/lib/dependencies/discover.service";
+import { CreateRequestContext, MikroORM } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { Command, Console } from "nestjs-console";
+import { v5 } from "uuid";
+
+import { Warning } from "./entities/warning.entity";
+import { WarningSeverity } from "./entities/warning-severity.enum";
+
+interface RootBlockEntityData {
+    tableName: string;
+    className: string;
+    rootBlockData: Array<{ block: Block; column: string }>;
+}
+
+@Injectable()
+@Console()
+export class WarningCheckerConsole {
+    constructor(
+        private readonly orm: MikroORM,
+        private readonly discoverService: DiscoverService,
+        private readonly entityManager: EntityManager,
+        @InjectRepository(Warning) private readonly warningsRepository: EntityRepository<Warning>,
+    ) {}
+
+    @Command({
+        command: "check-warnings",
+        description: "Checks for warnings",
+    })
+    @CreateRequestContext()
+    async execute(): Promise<void> {
+        // TODO: (in the next PRs) Check if data itself is valid in the database. (Maybe some data was put into database and is not correct or a migration was done wrong)
+        for (const data of this.groupRootBlockDataByEntity()) {
+            const { tableName, className, rootBlockData } = data;
+
+            const queryBuilderLimit = 100;
+            const baseQueryBuilder = this.entityManager.createQueryBuilder(className);
+            baseQueryBuilder
+                .select(["id", ...rootBlockData.map(({ column }) => column)])
+                .from(tableName)
+                .limit(queryBuilderLimit);
+            let rootBlocks: Array<{ [key: string]: BlockInput }> = [];
+            let offset = 0;
+
+            do {
+                const queryBuilder = baseQueryBuilder.clone();
+                queryBuilder.offset(offset);
+                rootBlocks = (await queryBuilder.getResult()) as Array<{ [key: string]: BlockInput }>;
+
+                for (const { column, block } of rootBlockData) {
+                    for (const rootBlock of rootBlocks) {
+                        const blockData = block.blockDataFactory(block.blockInputFactory(rootBlock[column]));
```

Comment:
This line is wrong:

- "input" is the thing that comes into the api using a mutation
- the data stored in database must be loaded differently, see here how this is done: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/blocks/root-block-type.ts#L53
- and the funny thing now is: QueryBuilder somehow uses the types defined in entities and does the conversion in RootBlockType already
- so you can simply skip this, you do already have a block object!

Reply:
Thank you! I can't explain how I came up with that. changed in: b2de862169669e83b4ff60bec28e69a1af3aeb75